src/resultset/resultset.c

```diff
@@ -90,17 +88,21 @@ ResultSet *NewResultSet(RedisModuleCtx *ctx, bool compact) {
 }
 
 // Initialize the user-facing reply arrays.
-void ResultSet_ReplyWithPreamble(ResultSet *set, QueryGraph *qg) {
-	if(set->exps == NULL || array_len(set->exps) == 0) {
+void ResultSet_ReplyWithPreamble(ResultSet *set, QueryGraph *qg, AR_ExpNode **columns) {
+	if(columns == NULL || array_len(columns) == 0) {
 		// Queries that don't form result sets will only emit statistics
 		RedisModule_ReplyWithArray(set->ctx, 1);
```

Comment:
wouldn't hurt setting data_emitted to false
also the name data_emitted is a bit misleading, I understand it as, actual records been emitted.